source/agora/utils/TxBuilder.d

```diff
@@ -227,9 +231,52 @@ public struct TxBuilder
         return this;
     }
 
+    /***************************************************************************
+
+        Sets the unlocker function to sign the inputs
+
+        If not set then the default unlocker using WellKnownKeys is used.
+
+        Params:
+            unlocker = function to sign the inputs of the transaction
+
+        Returns:
+            A reference to `this` to allow for chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) unlockSigner (Unlock
+        function (in Transaction tx, in OutputRef out_ref) @safe nothrow unlocker) return scope
+        @safe nothrow @nogc pure
+    {
+        this.unlocker = unlocker;
+        return this;
+    }
+
+    public static Unlock signWithGenesis (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow
+    {
+        auto sig = WK.Keys.Genesis.sign(tx);
+        return Unlock([ubyte(64)] ~ sig.toBlob()[]);
+    }
+
+    public static Unlock signWithSpecificKeyTrue (KeyPair key) (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow
+    {
+        auto sig = key.sign(tx);
+        return Unlock([ubyte(64)] ~ sig.toBlob()[] ~ [ubyte(OP.TRUE)]);
+    }
+
+    public static Unlock signWithSpecificKeyFalse (KeyPair key) (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow
+    {
+        auto sig = key.sign(tx);
+        return Unlock([ubyte(64)] ~ sig.toBlob()[] ~ [ubyte(OP.FALSE)]);
+    }
```

Comment:
What are those ?

Reply:
These are used in `Flash` tests. Shall I make it template function with `opcode` `ubyte`, or do you prefer I keep those functions in the `Flash` unit test?

Reply:
Well the `true` / `false` could just be a runtime parameter, no ?

Reply:
This is the function passed to the `txBuilder` for signing so I think the extra parameter would not be compatible.